context.js

```diff
@@ -144,7 +146,12 @@ class Context {
 
   unexpectedError (err) {
     this.errors.push(err)
-    this.output = String((err && err.stack) || err)
+    const message = typeof this.errorFormatter === 'function'
+      ? String(this.errorFormatter(err))
+      : String((err && err.stack) || err)
+    this.output = typeof this.styleUnexpectedError === 'function'
+      ? this.styleUnexpectedError(message)
+      : message
```

Comment:
I wish there was a way to do this where we didn't have to split style hook executions between different locations, but I can't think of a better way of doing it than this.

The only _potential_ problem I see is that this doesn't leverage the `all` style hook (as a final wrapper for all other styled strings), but it wasn't doing that before, and I'm not sure if a CLI author would expect the `all` hook to apply to a formatted error message or not (currently it only applies to help text and expected CLI messages). What do you think?

Reply:
I guess it depends what you're doing with `all`.  Possible use case:

```js
all: text => {
    // Example weird use case: always return output as a JSON blob containing lines as an array,
    // for consumption by some internal API.
    return JSON.stringify({ lines: text.split('\n') })
}
```

If this is how I was using sywac, then I _would_ want any output error messages to be run through it as well (although, in this case, I'm probably using `parse()` and could just examine `.errors` instead of the output).

My concern about just wrapping it with `all` is that you would potentially run afoul of `all` handers written just to handle the expected help text screen and not a generic error.  I almost feel you need to pass a second boolean option to tell the function whether this is a "help text" context or "unexpected error" context... but if you did that, and the user actually uses the boolean to make a decision, then I feel you could just as easily require them to put the logic in the unexpected error handler:

```js
all: text => {
    return JSON.stringify({ lines: text.split('\n') })
},
unexpectedError: text => {
    return JSON.stringify({ lines: text.split('\n') })
},
```

TLDR: This is a good point but I think I land on the side of "all does not apply to unexpected errors, because that's more flexible than the alternative".

Reply:
I think I agree with you. Thanks for the feedback!